benchmarks/benchmark_shared.c

```diff
@@ -116,3 +120,61 @@ int pingpong(void)
 
 	return 0;
 }
+
+int bandwidth(void)
+{
+	int ret, i, j;
+
+	ret = ft_sync();
+	if (ret)
+		return ret;
+
+	if (opts.dst_addr) {
+		for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+			if (i == opts.warmup_iterations)
+				ft_start();
+
+			for(j = 0; j < opts.window_size; j++) {
+				if (opts.transfer_size < fi->tx_attr->inject_size)
+					ret = ft_inject(opts.transfer_size);
+				else
+					ret = ft_post_tx(opts.transfer_size);
+				if (ret)
+					return ret;
+			}
+			ret = ft_get_tx_comp(tx_seq);
+			if (ret)
+				return ret;
+			ret = ft_rx(4);
+			if (ret)
+				return ret;
+		}
+	} else {
+		for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+			if (i == opts.warmup_iterations)
+				ft_start();
+
+			for(j = 0; j < opts.window_size; j++) {
+				ret = ft_post_rx(opts.transfer_size);
+				if (ret)
+					return ret;
+			}
```

Comment:
Yup, I agree. This was intentional, as this replicates the MPI benchmark behavior. For better or worse, this is how it has been measured for a while now. Some could argue that this represents a burst an application could cause.

In terms of overrunning the receiver, it can't overrun too much, since there is a synchronizing effect on a per window number transfers basis.